pulp_ansible/app/tasks/collections.py

```diff
@@ -969,6 +977,8 @@ async def run(self):
             else:
                 tasks.append(loop.create_task(self._find_all_collections()))
             await asyncio.gather(*tasks)
+            # Ensure PR 'total' is correct before stage finishes
+            pb.total = pb.done
```

Comment:
Should this be an assertion? What can go wrong here without being fatal?

Reply:
An assertion will cause the sync to fail pretty often with how I currently have it set up. For non-full syncs, the total typically overshoots the done number by a small amount. This is due to the fact that each item I add to the total (num of tasks spawned to process a collection version) does not equal the number of collection versions that gets put into the pipeline due to filtering.

Reply:
Hmm this makes me uneasy as we are not only having a running total, but also sounds like counting a different thing. Should we count these with separate progress bars instead?

Reply:
I'd guess in weird situations with aggressive filtering we'd end up with very different figures.

Reply:
We are estimating how many collection versions we are possibly adding to the repository, not sure what need there is for a second progress bar.

Reply:
We could but I see no point, and doing so gets away from the crux of the issue which is to implement a total count.

Reply:
I think this is the right thing to do. It's okay if the estimates are a little wonky while the task is running (progress bars fudge numbers all the time), but the total should match the completed at the end.

Reply:
So you two agree. I'll rest my case then.